app/views/RegisterView/index.tsx

```diff
@@ -0,0 +1,350 @@
+import React, { useLayoutEffect, useState } from 'react';
+import { Keyboard, Text, View } from 'react-native';
+import RNPickerSelect from 'react-native-picker-select';
+import parse from 'url-parse';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm, Controller } from 'react-hook-form';
+import { useDispatch } from 'react-redux';
+
+import { loginRequest } from '../../actions/login';
+import Button from '../../containers/Button';
+import FormContainer, { FormContainerInner } from '../../containers/FormContainer';
+import * as HeaderButton from '../../containers/HeaderButton';
+import LoginServices from '../../containers/LoginServices';
+import { FormTextInput } from '../../containers/TextInput';
+import { IBaseScreen } from '../../definitions';
+import I18n from '../../i18n';
+import { getShowLoginButton } from '../../selectors/login';
+import { OutsideParamList } from '../../stacks/types';
+import { useTheme } from '../../theme';
+import { showErrorAlert, isValidEmail } from '../../lib/methods/helpers';
+import { events, logEvent } from '../../lib/methods/helpers/log';
+import { Services } from '../../lib/services';
+import UGCRules from '../../containers/UserGeneratedContentRules';
+import { useAppSelector } from '../../lib/hooks';
+import PasswordTips from './PasswordTips';
+import getParsedCustomFields from './methods/getParsedCustomFields';
+import getCustomFields from './methods/getCustomFields';
+import styles from './styles';
+
+const passwordRules = /^(?!.*(.)\1{2})^(?=.*[a-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,24}$/;
+const validationSchema = yup.object().shape({
+	name: yup.string().min(1).required(),
+	email: yup.string().email().required(),
+	username: yup.string().min(1).required(),
+	password: yup.string().matches(passwordRules).required(),
+	confirmPassword: yup
+		.string()
+		.oneOf([yup.ref('password'), null])
+		.required()
+});
+
+interface IProps extends IBaseScreen<OutsideParamList, 'RegisterView'> {}
+
+const RegisterView = ({ navigation, route }: IProps) => {
+	const dispatch = useDispatch();
+	const { colors } = useTheme();
+	const { Accounts_CustomFields, Site_Url, Accounts_EmailVerification, Accounts_ManuallyApproveNewUsers, showLoginButton } =
+		useAppSelector(state => ({
+			...state.settings,
+			showLoginButton: getShowLoginButton(state),
+			Site_Url: state.settings.Site_Url as string,
+			Accounts_CustomFields: state.settings.Accounts_CustomFields as string
+		}));
+	const {
+		control,
+		handleSubmit,
+		setFocus,
+		getValues,
+		watch,
+		formState: { isValid, isDirty }
+	} = useForm({
+		mode: 'onChange',
+		defaultValues: {
+			name: '',
+			email: '',
+			password: '',
+			confirmPassword: '',
+			username: ''
+		},
+		resolver: yupResolver(validationSchema)
+	});
+	const password = watch('password');
+	const parsedCustomFields = getParsedCustomFields(Accounts_CustomFields);
```

Comment:
https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/6036/files#r1876431243

Reply:
got it, created.